### PR TITLE
Zio 2 support

### PR DESCRIPTION
**Pull request by Wosin**
# Description
PR to update ZIO version to 2.0.0 and and do all the necessary code changes to reflect that.
# Checklist

## Pull Request

* [ X] PR description is meaningful
* [ X] There are no more than a few commits to merge
* [ -] There are no more than a few lines of code to review

## Code style

* [ -] Maximum line length does not require horizontal scrolling in Github
* [ -] Compliant to FMT, ScalaStyle and ScalaFix

## Testing

* [ X] Unit tests for the added/modified code 
* [ X] Overall unit test coverage good
* [ X] Unit tests are passing

## Dependencies

* [ X] The safety plugin is happy
* [ X] New dependencies do not have vulnerabilities

## Security

* [ X] There is no known vulnerability being added to this project

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -41,7 +41,7 @@ lazy val rootSettings =
     dynver ~= (v => s"${sparkVersionSystem}_$v")
   )
 
-lazy val zioVersion = "1.0.16"
+lazy val zioVersion = "2.0.0"
```
Comment (leobenkel) [inline build.sbt:44]: should we be supporting zio 1.x and zio 2.x ? Like for spark versions ?

File: configLibs/Scallop/src/main/scala/com/leobenkel/zparkio/config/scallop/CommandLineArgumentScallop.scala [hunk 2]
```diff
@@ -59,21 +59,22 @@ object CommandLineArgumentScallop {
       subCommand: Option[String]
   ) extends Throwable
       with CommandLineArguments.Helper.HelpHandlerException {
-    private def print(msg: String): ZIO[Console, Throwable, Unit] = console.putStr(msg)
+    private def print(msg: String): ZIO[Console, Throwable, Unit] =
+      ZIO.consoleWith(console => console.printLine(msg))
```
Comment (leobenkel) [inline configLibs/Scallop/src/main/scala/com/leobenkel/zparkio/config/scallop/CommandLineArgumentScallop.scala:63]: `console` can be `_`

**Review by leobenkel — CHANGES_REQUESTED** [review]
Thanks ! It looks good !

File: build.sbt (hunk 1 above)
Comment (WosinDom) [reply build.sbt:44]: Hmm, we could do this from separate branches (like ZIO does) or separate code somehow but that's much more complex refactoring.

**Comment by leobenkel** [comment]
ScalaFmt needs to run

File: build.sbt (hunk 1 above)
Comment (leobenkel) [reply build.sbt:44]: hmm ok

Comment (Wosin) [reply build.sbt:44]: Unless, You know other way to make that work ofc :)

**Review by leobenkel — APPROVED** [review]

**Pull request merged**